src/FileSystem-Core/FileReference.class.st

```diff
@@ -253,18 +253,20 @@ FileReference >> isWritable [
 
 { #category : #utilities }
 FileReference >> lastFileFor: baseFileName extension: extension [
-  "Assumes a file is named using a version number encoded as '.' followed by digits
+	"Assumes a file is named using a version number encoded as '.' followed by digits
   preceding the file extension, e.g., games.22.ston
   Answer the file name with the largest number.
   If a version number is not found, raises an error"
 
-  "FileSystem workingDirectory lastFileFor: 'games' extension: 'ston'"
-
-    | files |
-    files := self childrenMatching: baseFileName , '.*.' , extension.
-    files ifEmpty: [ ^ self error: 'No file with number pattern' ].
-    ^ (files asSortedCollection: [ :a :b | a basename < b basename ]) last
+	"FileSystem workingDirectory lastFileFor: 'games' extension: 'ston'"
 
+	| files |
+	files := self childrenMatching: baseFileName , '.*.' , extension.
+	files ifEmpty: [ ^ self error: 'No file with number pattern' ].
+	^ (files
+		asSortedCollection:
+			[ :a :b | ((a basename splitOn: $.) at: 2) asNumber < ((b basename splitOn: $.) at: 2) asNumber ])
```

Comment:
the index (2) does expect that there is not another dot in the file name. It will not work for names like "file.name.001.jpg"

Reply:
actually, the code will cause error as soon as you have file name like file.notANumber.txt

Reply:
ok but the precondition is that there is no other . in the file name this is the same logic in the find next method. I will see how to fix it.